### DISCUSSION TITLE
Preview compile result in editor

### DISCUSSION THREAD

**Discussion by tocque** [reactwg/react-compiler#26] [Feature Requests]
Currently, the compilation of the React Compiler is completely a black box. Unless one views the compilation result, there is no way to know whether a certain piece of code is truly optimized. Is it possible to provide an editor plugin to mark in some form (such as Inlay Hints) how a certain piece of code will be optimized?

**Comment by orta** [comment]
This was noted as something they were exploring in the [most recent blog post on the React Compiler hitting Beta](https://react.dev/blog/2024/10/21/react-compiler-beta-release)

> Throughout this process, we also plan to prototype an IDE extension for React. It is still very early in research, so we expect to be able to share more of our findings with you in a future React Labs blog post.

Which I would expect would do this, it'll be interesting to see where that goes

**Comment by poteto** [comment]
Yeah we're working on an IDE extension. In the meantime, you can use the React DevTools which will show you a ✨ badge next to React Compiler optimized components.

**Reply by guillaumebrunerie** [reply]
> Yeah we're working on an IDE extension.

By "IDE", do you mean VSCode, or do you mean an extension that would work with several different IDEs, like a LSP server?

**Reply by poteto** [reply]
The latter (lsp) is what we're aiming for.

**Reply by blazejkustra** [reply]
While we are waiting for the official extension, would you consider sharing the source code for the VSCode extension that you showcased in [this](https://youtu.be/lvhPq5chokM?si=1FJclzPx5NImKHWc&t=429) talk, @poteto? I'd love to have ✨ badge next to the compiled components 😅

**Reply by blazejkustra** [reply]
I've always wanted to create a VSCode extension, so I decided to give it a go 😅 

Until the official IDE extension is out, you can check out [this extension](https://marketplace.visualstudio.com/items?itemName=blazejkustra.react-compiler-marker). It adds ✨ for optimized and 🚫 for non-optimized React components.

**Reply by colin969** [reply]
@blazejkustra Actual lifesaver.

I'm using it with an electron app, and since React Devtools hasn't really worked with it for years this is the only way I can get any feedback about what the compiler is doing at the moment.
